nixos/modules/hardware/fwenv.nix

```diff
@@ -0,0 +1,65 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.hardware.fwenv;
+in
+
+{
+  options.hardware.fwenv = {
```

Comment:
Not sure if it should go into the `hardware` namespace.

Reply:
BTW, `fwenv` seems to be a generic name for something that is related to U-Boot.

Reply:
`hardware` contains support for access to some hardware specific peripheries, as far as I can see. This is access to the some MTD device (commonly NOR) for U-Boot. The second place I was thinking about placing this was `boot` but honestly this is up to discussion.

`fwenv` comes from the file name, nothing more, nothing less. It might be more beneficial to name it `uboot.fwenv` or something like that if you want to explicitly relate that to U-Boot. In such case the `boot.uboot.env` path looks pretty nice.

Reply:
This is a complicated issue. This is not hardware, this is not the OS (NixOS), and is not *really* about the operating system boot.

So as far as namespacing, it's a non-trivial task :/.

Reply:
Moved.